ocaml/lambda/lambda.ml

```diff
@@ -539,6 +553,41 @@ let make_key e =
     Some (tr_rec Ident.empty e)
   with Not_simple -> None
 
+let join_actions_to_be_shared ~printer:_ lam1 lam2 =
```

Comment:
Why are we catching such a specific case rather than merge any lambda terms that differ only in read semantics?